common/quotas/global/algorithm/requestweighted.go

```diff
@@ -136,6 +136,8 @@ type (
 	// requests holds the running per-second running average request data for a single key from a single host,
 	// and the last time it was updated.
 	requests struct {
+		// TODO: add "first update" time, don't return to callers until it has passed 1(+?) update cycle as data is incomplete.
+
```

Comment:
I started building this - I think it's a good idea, but it'll mean changing a fair number of large tests because it's a meaningful semantic change.

But the way it works now is that after a ring change, it'll allow the very first hosts that update to get higher weights than they should for _one cycle_.  And then the next update cycle will correct them all.  Relatively low risk imo.